services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -148,6 +149,22 @@ open class HankeServiceImpl(
         return savedHanke
     }
 
+    /**
+     * Create application when no existing hanke. Autogenerates hanke and applies application to it.
+     */
+    @Transactional
+    override fun generateHankeWithApplication(
+        application: Application,
+        userId: String
+    ): HankeWithApplications {
+        val hanke = generateHankeFrom(application)
```

Comment:
Generating the hanke automatically from an application should only be allowed for johtoselvityshakemus, not any other kind of applications. This should be checked somewhere.

While we don't have other types of applications at the moment, it will be really easy to forget adding this check when we add other types of applications later, so the check should be added now.

Reply:
This method only accepts CableReportWithoutHanke so any other type of application cannot be used. Will that do?

Reply:
Yes, there's no longer any way to end up here with something other than a cable report application.